modules/percolator/src/internalClusterTest/java/org/elasticsearch/percolator/PercolatorQuerySearchIT.java

```diff
@@ -1064,79 +1151,90 @@ public void testPercolatorQueryViaMultiSearch() throws Exception {
+                        .setQuery(new PercolateQueryBuilder("query", "test", "6", null, null, null))
+                )
+                .get();
 
-        MultiSearchResponse.Item item = response.getResponses()[0];
-        assertHitCount(item.getResponse(), 2L);
-        assertSearchHits(item.getResponse(), "1", "4");
-        assertThat(item.getFailureMessage(), nullValue());
-
-        item = response.getResponses()[1];
-        assertHitCount(item.getResponse(), 2L);
-        assertSearchHits(item.getResponse(), "2", "4");
-        assertThat(item.getFailureMessage(), nullValue());
-
-        item = response.getResponses()[2];
-        assertHitCount(item.getResponse(), 4L);
-        assertSearchHits(item.getResponse(), "1", "2", "3", "4");
-        assertThat(item.getFailureMessage(), nullValue());
-
-        item = response.getResponses()[3];
-        assertHitCount(item.getResponse(), 1L);
-        assertSearchHits(item.getResponse(), "4");
-        assertThat(item.getFailureMessage(), nullValue());
-
-        item = response.getResponses()[4];
-        assertHitCount(item.getResponse(), 2L);
-        assertSearchHits(item.getResponse(), "2", "4");
-        assertThat(item.getFailureMessage(), nullValue());
-
-        item = response.getResponses()[5];
-        assertThat(item.getResponse(), nullValue());
-        assertThat(item.getFailureMessage(), notNullValue());
-        assertThat(item.getFailureMessage(), containsString("[test/6] couldn't be found"));
+            MultiSearchResponse.Item item = response.getResponses()[0];
+            assertHitCount(item.getResponse(), 2L);
+            assertSearchHits(item.getResponse(), "1", "4");
+            assertThat(item.getFailureMessage(), nullValue());
+
+            item = response.getResponses()[1];
+            assertHitCount(item.getResponse(), 2L);
+            assertSearchHits(item.getResponse(), "2", "4");
+            assertThat(item.getFailureMessage(), nullValue());
+
+            item = response.getResponses()[2];
+            assertHitCount(item.getResponse(), 4L);
+            assertSearchHits(item.getResponse(), "1", "2", "3", "4");
+            assertThat(item.getFailureMessage(), nullValue());
+
+            item = response.getResponses()[3];
+            assertHitCount(item.getResponse(), 1L);
+            assertSearchHits(item.getResponse(), "4");
+            assertThat(item.getFailureMessage(), nullValue());
+
+            item = response.getResponses()[4];
+            assertHitCount(item.getResponse(), 2L);
+            assertSearchHits(item.getResponse(), "2", "4");
+            assertThat(item.getFailureMessage(), nullValue());
+
+            item = response.getResponses()[5];
+            assertThat(item.getResponse(), nullValue());
+            assertThat(item.getFailureMessage(), notNullValue());
+            assertThat(item.getFailureMessage(), containsString("[test/6] couldn't be found"));
+        } finally {
+            if (response != null) {
```

Comment:
Lets back this one out please, this should be resolved by adjusting the ref counting on `MultiSearchResponse` in production so this becomes a single `.decRef()`, not by us doing this manually in tests :)

Reply:
Yeap, makes sense. Was wondering whether we want to eliminate as many test instances as possible in this round and then revisit explicitly `MultiSearchResponse` when we have the `decRef` properly implement. Reverted.